transformations/french_conjugation_transformation/transformation.py

```diff
@@ -0,0 +1,390 @@
+                            sent_list.append(sent_pron)
+                        # join the conjugated sentences in a final one
+                        final_sent = " ".join(conjug_sent)
+                        final_sent = sentence0 + " " + final_sent
+                        # delete the extra space
+                        final_sent = re.sub(" +", " ", final_sent)
+                        final_sent = final_sent.replace(" , ", ", ")
+                        final_sent = final_sent.replace("' ", "'").replace(
+                            " '", "'"
+                        )
+                        perturbed_texts.append(final_sent)
+                        return perturbed_texts
+                else:
+                    indexBefore = index_pronoun - 1
+                    if index_pronoun != 0:
+                        split_sentence = (
+                            sentence[:indexBefore]
+                            + "."
+                            + sentence[indexBefore:]
+                        )
+                        splited_sentences = nltk.sent_tokenize(split_sentence)
+                        sent_list = []
+                        conjug_sent = []
+
+                        for i in range(len(splited_sentences)):
+                            sent_pron = [
+                                splited_sentences[i],
+                                pronouns_final_list[i],
+                            ]
+                            conjugated = conjugate(
+                                splited_sentences[i],
+                                pronouns_final_list[i],
+                                tense,
+                            )
+                            # delete the "." added in the split sentence process
+                            conjugated = conjugated.replace(".", "")
+                            conjug_sent.append(conjugated)
+                            sent_list.append(sent_pron)
+                        # join the conjugated sentences in a final one
+                        final_sent = " ".join(conjug_sent)
+                        # delete the extra space
+                        final_sent = re.sub(" +", " ", final_sent)
+                        final_sent = final_sent.replace(" , ", ", ")
+                        final_sent = final_sent.replace("' ", "'").replace(
+                            " '", "'"
+                        )
+
+                        perturbed_texts.append(final_sent)
+                        return perturbed_texts
+
+
+class Conjugation_transformation(SentenceOperation):
+
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+    ]
+    languages = ["fr"]
+    KEYWORD = [
+        "lexical",
+        "model-based",
+        "highly-meaning-preserving",
+        "high-coverage",
+    ]
+
+    # The transormation tense is specified here
+    def __init__(self, tense):
+        super().__init__()
+        assert tense in ["Futur", "Imparfait"]
```

Comment:
Does this correspond to the documentation? You state more tenses there than here?

Reply:
The tenses specified in the read me can also be passed since they are supported by the mlcnonjug library. I didn't pass them in my function, I had issues with utf-8 encoding when I specify a tense containing an accent (looks like the test json file isn't opened with the right encoding)